### DISCUSSION TITLE
Failing to lookup all public npm packages

### DISCUSSION THREAD

**Discussion by dylanpomeroy** [renovatebot/renovate#14499] [Request Help]
Hello 👋 

I am trying to setup renovate for my team's project. I have it configured to look only at our main `package.json` at root which contains about 200 public npm packages (and some private ones).

I am getting warnings for every public npm package saying that they failed lookup.

When checking the renovate logs, it appears the bot is looking them up using a private repository we have configured for our private packages in `.yarnrc` and `.npmrc` files, but this is not setup in the `renovate.json` config. I will include the log for one at the end.

While we do have a private GemFury registry to add, I am planning on doing that in the future and aiming to get our public dependencies managed by Renovate. Is there something I'm missing in the config? Docs give lots of info on setting up private lookups, but they do not cover details on public lookup.

Thanks in advance!

Log for package lookup failure:
```
DEBUG: Dependency lookup failure: not found
{
  "err": {
    "name": "HTTPError",
    "code": "ERR_NON_2XX_3XX_RESPONSE",
    "timings": {
      ...
    },
    "message": "Response code 404 (Not Found)",
    "stack": "HTTPError: Response code 404 (Not Found)\n    at Request.<anonymous> (/home/ubuntu/renovateapp/node_modules/got/dist/source/as-promise/index.js:117:42)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (internal/process/task_queues.js:95:5)",
    "options": {
      "headers": {
        "user-agent": "Renovate Bot (GitHub App 2740)",
        "accept": "application/json",
        "accept-encoding": "gzip, deflate, br"
      },
      "url": "https://npm-proxy.fury.io/wave/@apollo%2Fclient",
      "hostType": "npm",
      "username": "",
      "password": "",
      "method": "GET",
      "http2": false
    },
    "response": {
      "statusCode": 404,
      "statusMessage": "Not Found",
      "body": {
        "error": "bad_request",
        "reason": "Repo not found"
      },
      "headers": {
        "server": "Cowboy",
        "connection": "close",
        "date": "Thu, 03 Mar 2022 15:38:25 GMT",
        "content-length": "50",
        "content-type": "text/plain; charset=utf-8",
        "via": "1.1 vegur"
      },
      "httpVersion": "1.1"
    }
  },
  "packageName": "@apollo/client"
}
DEBUG: Failed to look up dependency @apollo/client (@apollo/client)
```

**Comment by dylanpomeroy** [comment]
This is my current renovate.json config for reference:
```
{
  "extends": [
    "config:base",
    ":enableRenovate",
    ":separateMultipleMajorReleases",
    ":combinePatchMinorReleases",
    ":renovatePrefix",
    ":automergeDisabled"
  ],
  "ignorePaths": [".yarn/**", ".circleci/**", "e2e-tests/**", ".nvmrc"],
  "packageRules": [
    {
      "matchPackageNames": ["jest"],
      "enabled": false
    }
  ]
}

```

**Comment by rarkins** [comment]
Renovate reads/follows `.npmrc` files. If you have them in your repo, then it makes sense that Renovate will honor the instructions there. 

If you want to force override your `.npmrc` then I suggest configuring `"npmrc": "registry=https://registry.npmjs.org"`

**Comment by dylanpomeroy** [comment]
@rarkins that fixed my issue, thanks a bunch!

Out of curiosity, do you know why Renovate didn't pick up on public repositories with a private one configured in the `.npmrc`? We have the definition because a couple of our dependencies are private, but `npm install` and `yarn install` runs are able to do public lookups with that config active:
```
always-auth=true
registry=https://npm-proxy.fury.io/[redacted]/
//npm-proxy.fury.io/[redacted]/:_authToken=[redacted]
```

I wonder if there is a proper way to configure this in `.npmrc`, or at least a way that Renovate assumes it is setup upon parsing

**Reply by rarkins** [reply]
`registry=https://npm-proxy.fury.io/[redacted]/` means "Use this private fury registry for every single package including public ones"

If npm and yarn are working then it's because you've configured them with credentials - they are still going through fury for every request.

Naturally, Renovate cannot query anything at all from your private registry unless you grant it the credentials necessary.
